### DISCUSSION TITLE
Including first & last pages with pagy_nav?

### DISCUSSION THREAD

**Discussion by excid3** [ddnexus/pagy#694] [Q&A] [answered]
The docs show `pagy_nav` rendering this which always includes a gap and the first / last pages to make it easier to jump back to the start or end. Plus it's a helpful reference for how large the results are.

![image](https://github.com/ddnexus/pagy/assets/67093/9220adfd-85c4-4e2f-904b-3c393e4e77c2)

However, I get this with `pagy_nav` with an ActiveRecord relation.
<img width="377" alt="Screenshot 2024-05-02 at 4 13 12 PM" src="https://github.com/ddnexus/pagy/assets/67093/a7a27924-a5e3-4418-aa4c-5cfc0fcfb7ad">

Is there a way to have `pagy_nav` render the first / last pages still?

**Comment by benkoshy** [comment, marked as answer]
> Is there a way to have pagy_nav render the first / last pages still?

Yes.

If all you want is to render the first and last pages in addition to the others, all the time, you can do this by changing the `size` array:

```rb
pagy = Pagy.new count: 1000, page: 10, size: [1, 4, 4,1] # etc
pagy.series
#=>[1, :gap, 6, 7, 8, 9, "10", 11, 12, 13, 14, :gap, 50]

# so you may want to do something like in your FoosController.rb
@pagy, @foos = pagy(Foo.all, size: [1, 4, 4, 1])           

# or set a default in your pagy.rb initialiser if you want it globally set.
Pagy::DEFAULT[:size]    = [1, 4, 4,1]

```

## Reference
https://ddnexus.github.io/pagy/docs/how-to/#control-the-page-links and checkout the session on "classic nav".

### Try it out:

Run the interactive demo from your terminal:

```sh
bundle exec pagy demo
```

and point your browser to `http://0.0.0.0:8000`

![image](https://github.com/ddnexus/pagy/assets/15097447/94cec510-295e-4c25-ab4c-1f34d33f17be)

(see [pagy playground](https://ddnexus.github.io/pagy/playground))

### Targeting Classes
Looks like you're using bootstrap (?) but for anyone else, who wants to [target the first / last link via css](https://ddnexus.github.io/pagy/docs/api/stylesheets/#html-structure).  

> You can target the previous and next links by using pagy a:first-child and pagy a:last-child pseudo classes


## Credit
@ddnexus who suggested the references.

**Reply by excid3** [reply]
Ah, that's what `size` does. 👍

**Reply by ddnexus** [reply]
@excid3 FYI, v8.6 implemented a faster default bar that is more symmetrical and stable in size and includes the first and last pages when needed. The default is `size: 7` (i.e. 7 page/gaps slots) and looks like this:

<img width="321" alt="Screen Shot 2024-06-28 at 4 54 29 PM" src="https://github.com/ddnexus/pagy/assets/100721/afebf6a7-c4fe-4ae2-8e1f-5c751540f518">

If you want a wider bar, just increase the value. Odd numbers work better because the current page is placed in the center. E.g. `size: 11`:

<img width="475" alt="Screen Shot 2024-06-28 at 5 00 39 PM" src="https://github.com/ddnexus/pagy/assets/100721/f85ae8bb-037c-4112-9fb6-4d2b521ccc55">

HTH
